src/Sylius/Behat/Context/Setup/PromotionContext.php

```diff
@@ -0,0 +1,111 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Setup;
+
+use Behat\Behat\Context\Context;
+use Doctrine\Common\Persistence\ObjectManager;
+use Sylius\Component\Core\Model\PromotionInterface;
+use Sylius\Component\Core\Test\Factory\TestPromotionFactoryInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Promotion\Factory\ActionFactoryInterface;
+use Sylius\Component\Promotion\Model\ActionInterface;
+use Sylius\Component\Promotion\Model\RuleInterface;
+use Sylius\Component\Promotion\Repository\PromotionRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class PromotionContext implements Context
+{
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var ActionFactoryInterface
+     */
+    private $actionFactory;
+
+    /**
+     * @var TestPromotionFactoryInterface
+     */
+    private $testPromotionFactory;
+
+    /**
+     * @var PromotionRepositoryInterface
+     */
+    private $promotionRepository;
+
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param ActionFactoryInterface $actionFactory
+     * @param TestPromotionFactoryInterface $testPromotionFactory
+     * @param PromotionRepositoryInterface $promotionRepository
+     * @param ObjectManager $objectManager
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        ActionFactoryInterface $actionFactory,
+        TestPromotionFactoryInterface $testPromotionFactory,
+        PromotionRepositoryInterface $promotionRepository,
+        ObjectManager $objectManager
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->actionFactory = $actionFactory;
+        $this->testPromotionFactory = $testPromotionFactory;
+        $this->promotionRepository = $promotionRepository;
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * @Given there is a promotion :promotionName
+     */
+    public function thereIsPromotion($promotionName)
+    {
+        $promotion = $this->testPromotionFactory->create($promotionName);
+        $promotion->addChannel($this->sharedStorage->get('channel'));
+
+        $this->promotionRepository->add($promotion);
+        $this->sharedStorage->set('promotion', $promotion);
+    }
+
+    /**
+     * @Given /^it gives "(?:€|£|\$)([^"]+)" fixed discount to every order$/
+     */
+    public function itGivesFixedDiscountToEveryOrder($amount)
+    {
+        $currentPromotion = $this->sharedStorage->get('promotion');
+
+        $action = $this->actionFactory->createFixedDiscount($this->getPriceFromString($amount));
+        $currentPromotion->addAction($action);
```

Comment:
I think that method ``createFixedDiscount`` should only create fixed discount action, not also add it to promotion. We could of course create ``createFixedDiscountForPromotion`` method...
However I'm not sure about it. It could leads to some confusion when reading ``itGivesFixedDiscountToEveryOrder`` method, as it would create action and flush ``objectManager``... and first question could be "why? we've just created ``$action``, why flushing manager?". Adding ``$action`` to ``$promotion`` at this level it is clear what is it used for and why ``objectManager`` should be flushed.